src/compiler/checker.ts

```diff
@@ -10664,6 +10740,11 @@ namespace ts {
             }
 
             forEach(node.members, checkSourceElement);
+            
+            // Classes containing abstract members must be marked abstract
+            if (!(node.flags & NodeFlags.Abstract) && forEach(node.members, (element: ClassElement) => element.flags & NodeFlags.Abstract)) {
+                error(node, Diagnostics.Classes_containing_abstract_functions_must_be_marked_abstract);

```

Comment:
Changed.

In general, should we refer to member functions as "methods" or "member functions"? Or is this insubstantial for us?